app/models/node.rb

```diff
@@ -877,6 +877,42 @@ def self.questions
         .group('node.nid')
   end
 
+  # finds nodes by tag name, user id, and optional node type
+  def self.find_by_tag_and_author(tagname, user_id, type = 'note')
```

Comment:
Interesting -- very cool! @cesswairimu and I are looking here and I thought we already had a way to do this... maybe? In the controller for https://publiclab.org/tag/water-quality/author/mimiss ? Just wondering where this new code is being used, and if it is possibly redundant with that route? Thanks for helping me understand, this is a big awesome PR!

Reply:
I didn't see anything that matched, but I'll take a look at that page and see what it uses!

Reply:
So yes, there is similar logic in TagController#show_for_author, but the variables are passed in via the URL. I'll have to figure out how to refactor my code to use this.

Reply:
So the issue with this is that I can pull the logic out into a separate function to use in TagController#show_for_author but the problem is that this function doesn't just generate one list of notes, it creates one for notes, questions, answered questions, wikis and nodes. Several of these lists are referenced in other view templates... basically trying to clean it up breaks a bunch of other things.